prql-php/src/Compiler.php

```diff
@@ -75,113 +68,179 @@ function __construct(?string $lib_path = null)
-     * be compiled.
+     * @return Result compilation result containing RQ serialized as JSON.
      * @api
      */
-    function plToRQ(string $pl_json): string
+    function plToRQ(string $pl_json): Result
     {
-        if (!$prql_query) {
+        if (!$pl_json) {
             throw new \InvalidArgumentException("No query given.");
         }
 
-        $out = str_pad("", 1024);
-        if ($this->_libprql->pl_to_rq($pl_json, $out) !== 0) {
-            throw new \InvalidArgumentException("Could not convert PL.");
-        }
+        $res = $this->_libprql->pl_to_rq($pl_json);
 
-        return trim($out);
+        return $this->convert_result($res);
     }
 
     /**
      * Converts RQ to SQL.
      *
      * @param string $rq_json PL in JSON format.
      *
-     * @return string SQL string.
-     * @throws \InvalidArgumentException If no query is given or the query cannot
-     * be compiled.
+     * @return Result compilation result containing SQL query.
      * @api
      */
-    function rQToSql(string $rq_json): string
+    function rqToSQL(string $rq_json, ?Options $options = null): Result
     {
-        if (!$prql_query) {
+        if (!$rq_json) {
             throw new \InvalidArgumentException("No query given.");
         }
 
-        $out = str_pad("", 1024);
-        if ($this->_libprql->rq_to_sql($rq_json, $out) !== 0) {
-            throw new \InvalidArgumentException("Could not convert RQ.");
+        $ffi_options = $this->options_init($options);
+
+        $res = $this->_libprql->rq_to_sql($rq_json, $out);
+
+        $this->options_destroy($ffi_options);
+
+        return $this->convert_result($res);
+    }
+
+    private function options_init(?Options $options = null) {
+        if ($options === null) {
+            $options = new Options();
+        }
+
+        $ffi_options = $this->_libprql->new("struct Options");
+        $ffi_options->format = $options->format;
+        $ffi_options->signature_comment = $options->signature_comment;
+
+        if (isset($options->target)) {
+            $len = strlen($options->target) + 1;
+            $ffi_options->target = \FFI::new("char[$len]", false);
+            \FFI::memcpy($ffi_options->target, $options->target, $len - 1);
+        }
+
+        return $ffi_options;
+    }
+
+    private function options_destroy($ffi_options) {
+        if (!\FFI::isNull($ffi_options->target)) {
+            \FFI::free($ffi_options->target);
         }
+        unset($ffi_options);
+    }
+
+    private function convert_result($ffi_res): Result {
+        $res = new Result();
+
+        // convert string
+        $res->output = $this->convert_string($ffi_res->output);
+
+        $res->messages = array();
+        for ($i = 0; $i < $ffi_res->messages_len; $i++) {
+            $res->messages[$i] = $this->convert_message($ffi_res->messages[$i]);
+        }
+
+        // free the ffi_result
+        $this->_libprql->result_destroy($ffi_res);
+        return $res;
```

Comment:
Return statement on separate line.
```suggestion

        return $res;
```